websocket/connection.go

```diff
@@ -92,6 +92,25 @@ func NewDurableSendingConnection(target string, logger *zap.SugaredLogger) *Mana
 	return NewDurableConnection(target, nil, logger)
 }
 
+// NewDurableSendingConnectionGuaranteed creates a new websocket connection
+// that can only send messages to the endpoint it connects to. It returns
+// the connection if the connection can be established within the given
+// `duration`. Otherwise it returns nil.
+//
+// The connection will continuously be kept alive and reconnected
+// in case of a loss of connectivity.
+func NewDurableSendingConnectionGuaranteed(target string, duration time.Duration, logger *zap.SugaredLogger) *ManagedConnection {
+	c := NewDurableConnection(target, nil, logger)
+
+	if err := wait.PollImmediate(5*time.Millisecond, duration, func() (bool, error) {
```

Comment:
Can we achieve this without a poll potentially? :thinking:

In broad strokes I'm thinking about adding a channel that is closed when the first connect is successful.

Reply:
done